tools/autograd/gen_variable_type.py

```diff
@@ -80,6 +80,28 @@
     '_coalesced_',
 }
 
+# When a function modifies its input tensors, it should never change the the input tensors'
+# underlying c10::TensorImpl pointers or c10::Storage pointers.
+#
+# The following code templates
```

Comment:
why aren't these just all together, then you can just put "#END CHECKS" instead of maintaining a list that is 100% going to go out of date.